analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/PubliclyWritableDirectoriesBase.cs

```diff
@@ -0,0 +1,133 @@
+
+        protected string[] InsecureEnvironmentVariables { get; } = { "tmp", "temp", "tmpdir" };
+
+        protected PubliclyWritableDirectoriesBase(IAnalyzerConfiguration configuration, ResourceManager rspecResources) : base(configuration) =>
+            rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, rspecResources).WithNotConfigurable();
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                Language.GeneratedCodeRecognizer,
+                c =>
+                {
+                    if (!IsEnabled(c.Options))
+                    {
+                        return;
+                    }
+
+                    var node = c.Node;
+                    if (Language.Syntax.NodeStringTextValue(node) is { } stringValue
+                        && PubliclyWritableDirectoriesRegexMatcher.IsSensitiveDirectoryUsage(stringValue))
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, node.GetLocation()));
+                    }
+                },
+                Language.SyntaxKind.StringLiteralExpression,
+                Language.SyntaxKind.InterpolatedStringExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                Language.GeneratedCodeRecognizer,
+                c =>
+                {
+                    if (!IsEnabled(c.Options))
+                    {
+                        return;
+                    }
+
+                    var node = c.Node;
+                    if (node is TInvocationExpression invocation
+                        && (IsGetTempPathAssignment(invocation, KnownType.System_IO_Path, "GetTempPath", c.SemanticModel)
+                            || IsInsecureEnvironmentVariableRetrieval(invocation, KnownType.System_Environment, "GetEnvironmentVariable", c.SemanticModel)))
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, node.GetLocation()));
+                    }
+                },
+                Language.SyntaxKind.InvocationExpression);
+        }
+
+        private static class PubliclyWritableDirectoriesRegexMatcher
+        {
+            private const string TwoPartPatternWithVariableSlash = @"^[\/\\]{0}[\/\\]{1}([\/\\]|$)";
+            private const string TwoPartPatternWithForwardSlash = @"^\/{0}\/{1}(\/|$)";
+            private const string ThreePartPatternWithForwardSlash = @"^\/{0}\/{1}\/{2}(\/|$)";
+            private const string TmpRegex = @"^([a-zA-z]:)?[\\\/]?[\/\\]tmp([\/\\]|$)";
+            private const string TempRegex = @"^([a-zA-z]:)?[\\\/]?[\/\\]temp([\/\\]|$)";
+            private const string WindowsTempRegex = @"^([a-zA-z]:)?[\\\/]?[\\\/]windows[\\\/]temp([\\\/]|$)";
+            private const string UserProfileRegex = @"%USERPROFILE%\\AppData\\Local\\Temp([\\\/]|$)";
```

Comment:
These are not regex objects, but patterns
```suggestion
            private const string UserProfilePattern = @"%USERPROFILE%\\AppData\\Local\\Temp([\\\/]|$)";
```
or
```suggestion
            private const string UserProfile = @"%USERPROFILE%\\AppData\\Local\\Temp([\\\/]|$)";
```